src/bin/client.rs

```diff
@@ -1,64 +1,95 @@
-use colored::*;
-use futures::SinkExt;
-use std::io;
-use std::io::Write;
-use tokio::net::TcpStream;
-use tokio::stream::StreamExt;
-use tokio_util::codec::{Framed, LengthDelimitedCodec};
+use clap::clap_app;
+use serde::{Deserialize, Serialize};
+mod protocol;
 
-static LOCAL_REMITS: &str = "localhost:4242";
+fn main() {
+    let matches = clap_app!(myapp =>
+        (version: "0.1")
+        (about: "Interact with remits")
+        (@arg debug: -d ... "Sets the level of debugging information")
+        (@subcommand log_add =>
+            (about: "Add log")
+            (@arg log_name: -n +required +takes_value "Log name to add")
+        )
+        (@subcommand log_list =>
+            (about: "List logs")
+        )
+        (@subcommand log_del =>
+            (about: "Delete log")
+            (@arg log_name: -n +required +takes_value "Log name to delete")
+        )
+        (@subcommand log_show =>
+            (about: "Show metadata of log")
+            (@arg log_name: -n +required +takes_value "Log name to see metadata")
+        )
+        (@subcommand msg_add =>
+            (about: "Add message to log")
+            (@arg msg: -m +takes_value "Value of msg to add")
+        )
+        (@subcommand itr_add =>
+            (about: "Add itr to log")
+            (@arg log: -l +required +takes_value "Value of log to add itr")
+            (@arg itr_name: -n +required +takes_value "choose itr name")
```

Comment:
The actual request now uses the full word `iterator_name` and refers to the command as `Iterator Add`.  It's more typing, but I think it'll be easier for people to learn about and pick up.  Mind using the same arg name here as in the docs/request?  The shortform of the flag `-n` should be short enough if it's tedious to type.